src/Sylius/Bundle/CoreBundle/Sitemap/Provider/ProductUrlProvider.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace Sylius\Bundle\CoreBundle\Sitemap\Provider;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Bundle\CoreBundle\Sitemap\Factory\SitemapUrlFactoryInterface;
+use Sylius\Bundle\CoreBundle\Sitemap\Model\SitemapUrlInterface;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ProductUrlProvider implements UrlProviderInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $repository;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SitemapUrlFactoryInterface
+     */
+    private $sitemapUrlFactory;
+
+    /**
+     * @var array
+     */
+    private $urls;
+
+    /**
+     * @param RepositoryInterface $repository
+     * @param RouterInterface $router
```

Comment:
Aligning expressions like that has no sense. If we remove the longest one or add new the lognest one, we have to change all the aligned expressions. Moreover, people read horizontally, not vertically, and table structure isn't helpful at all.